#! /bin/sh /usr/share/dpatch/dpatch-run
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Web config utility should use /var/lib/phpmyadmin as the save path.

@DPATCH@

--- phpmyadmin-2.10.0.2.orig/scripts/setup.php
+++ phpmyadmin-2.10.0.2/scripts/setup.php
@@ -1260,13 +1260,16 @@
     return FALSE;
 }
 
+define('WEBCONFIG_DIR', '/var/lib/phpmyadmin/');
+define('WEBCONFIG_FILE', WEBCONFIG_DIR . 'config.inc.php');
+
 if ($action != 'download') {
     // Check whether we can write to configuration
     $fail_dir = FALSE;
-    $fail_dir = $fail_dir || !is_dir('./config/');
-    $fail_dir = $fail_dir || !is_writable('./config/');
-    $fail_dir = $fail_dir || (file_exists('./config/config.inc.php') && !is_writable('./config/config.inc.php'));
-    $config = @fopen('./config/config.inc.php', 'a');
+    $fail_dir = $fail_dir || !is_dir(WEBCONFIG_DIR);
+    //$fail_dir = $fail_dir || !is_writable(WEBCONFIG_DIR);
+    $fail_dir = $fail_dir || (file_exists(WEBCONFIG_FILE) && !is_writable(WEBCONFIG_FILE));
+    $config = @fopen(WEBCONFIG_FILE, 'a');
     $fail_dir = $fail_dir || ($config === FALSE);
     @fclose($config);
 }
@@ -1313,7 +1316,7 @@
         <?php
         break;
     case 'save':
-        $config = @fopen('./config/config.inc.php', 'w');
+        $config = @fopen(WEBCONFIG_FILE, 'w');
         if ($config === FALSE) {
             message('error', 'Could not open config file for writing! Bad permissions?');
             break;
@@ -1324,7 +1327,7 @@
             message('error', 'Could not write to config file! Not enough space?');
             break;
         } else {
-            message('notice', 'Configuration saved to file config/config.inc.php in phpMyAdmin top level directory, copy it to top level one and delete directory config to use it.', 'File saved');
+            message('notice', 'Configuration saved.', 'File saved');
         }
         unset($r, $s);
         fclose($config);
@@ -1334,7 +1337,7 @@
             message('error', 'Reading of configuration disabled because of permissions.');
             break;
         }
-        $new_cfg = load_config('./config/config.inc.php');
+        $new_cfg = load_config(WEBCONFIG_FILE);
         if (!($new_cfg === FALSE)) {
             $configuration = $new_cfg;
         }
